uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -402,17 +414,17 @@ define([
        * @return {Object} attributes
        */
       this._initRequestAttributes = function () {
-        var attributes;
+        var attributes = {};
 
         //if set indicates self leaverequest is either being managed or edited
         if (this.directiveOptions.leaveRequest) {
-          //get a clone so self it is not the same reference as passed from callee
           //_.deepClone or angular.copy were not uploading files correctly
           attributes = this.directiveOptions.leaveRequest.attributes();
-        } else {
-          attributes = {
-            contact_id: this.directiveOptions.contactId
-          };
+
+          this.contact = { contact_id: attributes.contact_id };
```

Comment:
i'm not familiar with this logic, but again why do you need `this.contact` at all, can't you directly set `attributes.contact_id` without relying on `this.contact`